emailproxy.py

```diff
@@ -337,9 +344,158 @@ def add_account(username):
+                        if secret_id.startswith('arn:'):
+                            Log.error('Failed to store OAuth 2.0 tokens in AWS Secret "%s"' % secret_id, '- the Secret',
+                                      'does not exist, and it is not possible to create a secret with a specific ARN')
+                            AppConfig._aws_secrets_debug_log_boto3_error(err_put_secret)
+
+                        else:
+                            Log.error('AWS Secret "%s"' % secret_id, 'does not exist - attempting to create it')
+                            AppConfig._aws_secrets_debug_log_boto3_error(err_put_secret)
+
+                            try:
+                                aws_client.create_secret(Name=secret_id, ForceOverwriteReplicaSecret=False)
+
+                            except botocore.exceptions.ClientError as err_create_secret:
+                                if err_create_secret.response['Error']['Code'] == 'AccessDeniedException':
+                                    Log.error('Failed to store OAuth 2.0 tokens in AWS Secret "%s"' % secret_id,
+                                              '- access denied: does the IAM user have the',
+                                              '"secretsmanager:CreateSecret" permission?')
+                                else:
+                                    Log.error('Failed to store OAuth 2.0 tokens in AWS Secret "%s"' % secret_id,
+                                              '- attempt to create failed with unexpected error; see the proxy\'s',
+                                              'debug mode log for details')
+                                AppConfig._aws_secrets_debug_log_boto3_error(err_create_secret)
+
+                            else:
+                                Log.info('Created AWS Secret "%s"' % secret_id)
+                                try:
+                                    aws_client.put_secret_value(SecretId=secret_id,
+                                                                SecretString=json.dumps(aws_secrets[secret_id]))
+
+                                except botocore.exceptions.ClientError as err_put_secret_after_create:
+                                    if err_put_secret_after_create.response['Error']['Code'] == 'AccessDeniedException':
+                                        Log.error('Failed to store OAuth 2.0 tokens in AWS Secret "%s"' % secret_id,
+                                                  '- access denied: does the IAM user have the',
+                                                  '"secretsmanager:PutSecretValue" permission?')
+                                    else:
+                                        # TODO: missing format string
+                                        Log.error('Failed to store OAuth 2.0 tokens in AWS Secret "%s"' % secret_id,
+                                                  '- unexpected error; see the proxy\'s debug mode log for details')
```

Comment:
```suggestion
                                        Log.error('Failed to store OAuth 2.0 tokens in AWS Secret "%s"' % secret_id,
                                                  '- unexpected error; see the proxy\'s debug mode log for details')
```

Removing stale TODO comment